org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/utils/MarkupContentFactory.java

```diff
@@ -57,4 +59,33 @@ public static MarkupContent createMarkupContent(String value, String preferredKi
 		}
 		return content;
 	}
+
+	/**
+	 * Create the markup content according the given markup kind and the capability
+	 * of the client.
+	 * 
+	 * @param values         the list of documentation values
+	 * @return the markup content according the given markup kind and the capability
+	 *         of the client.
+	 */
+	public static MarkupContent creatMarkupContent(List<String> values, IMarkupKindSupport request) {
+		String kind = request.canSupportMarkupKind(MarkupKind.MARKDOWN) ? MarkupKind.MARKDOWN : MarkupKind.PLAINTEXT;
+		if (values.size() ==1) {
+			return new MarkupContent(kind, values.get(0));
+		}
+		StringBuilder retValue = new StringBuilder();
+		if (kind.equals(MarkupKind.MARKDOWN)) {
+			for (String value : values) {
+				retValue.append(value);
+				retValue.append("___");

```

Comment:
Do this test here if (kind.equals(MarkupKind.MARKDOWN)) {

Reply:
I think retValue.append(System.lineSeparator() ); must be done for all kind, no?

Reply:
Yes, you are right. I will modify it.
For the test, it only test when kind is markdown. should I add test for plaintext kind?

Reply:
we have not this kind of test. I think it's fine now. But please merge your 3 commits in once and after that I think we will able to merge your PR